src/text/line.rs

```diff
@@ -451,6 +451,40 @@ impl<'a> Line<'a> {
     pub fn iter_mut(&mut self) -> std::slice::IterMut<Span<'a>> {
         self.spans.iter_mut()
     }
+
+    /// Returns a line that's truncated corresponding to it's alignment and result width
+    #[must_use = "method moves the value of self and returns the modified value"]
+    pub fn truncated(&'a self, result_width: u16) -> Self {
+        let mut truncated_line = Line::default();
+        let width = self.width() as u16;
+        let mut offset = match self.alignment {
+            Some(Alignment::Center) => (width.saturating_sub(result_width)) / 2,
+            Some(Alignment::Right) => width.saturating_sub(result_width),
+            _ => 0,
+        };
+        let mut x = 0;
+        for span in &self.spans {
+            let span_width = span.width() as u16;
+            if offset >= span_width {
+                offset -= span_width;
+                continue;
```

Comment:
It might be a good idea to add a test case where this code is relevant. I see myself in the future touching and breaking code specifics like this accidentally.